### PR TITLE
feat: STAKE-827: track additional pooled staking events

### PR DESCRIPTION
**Pull request by Matt561**
## **Description**
Adds additional event tracking for pooled-staking tooltips and currency switching.

## **Related issues**

Jira Ticket: [STAKE-827: Track more events in Mixpanel](https://consensyssoftware.atlassian.net/browse/STAKE-827)

## **Manual testing steps**

**note: there's an ongoing issues with tracking events on local dev. Results may vary.**

Prerequisite: Add `export MM_POOLED_STAKING_UI_ENABLED=true` to your local `.js.env` file

Testing Currency Switch Events
1. Click on Ethereum in assets list
2. Click "Stake", "Stake More", or "Unstake" buttons to navigate to the input screen
3. Switch the currency between native and fiat near the top of the screen

Testing Tooltip Events
1. Click on Ethereum in assets list
2. Click on available tooltip icons: 
  - Annual rate in the "Your earnings" section within ETH details page
  - Learn more tooltip on the staking input screen
  - Various tooltips on the stake and unstake confirmation views
## **Screenshots/Recordings**
N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by amitabh94 — APPROVED** [review]
LGTM!

**Review by georgewrmarshall — APPROVED** [review]
Approving for @MetaMask/design-system-engineers

File: app/components/UI/Stake/utils/metaMetrics/withMetaMetrics.ts
Comment (Cal-L) [inline app/components/UI/Stake/utils/metaMetrics/withMetaMetrics.ts:1]: What's the added benefit in having this wrapper vs tracking events using `trackEvent` and `createEventBuilder` from `useMetrics` and/or `withMetricsAwareness`?

**Review by Cal-L — APPROVED** [review]
LGTM - Let's follow up + create ticket for either removing `withMetaMetrics` so we won't have to manage another piece and/or update the existing `useMetrics` and `withMetricsAwareness` code

**Pull request merged**